source/algorithms.tex

```diff
@@ -10755,6 +10767,123 @@
 where the result of the division is truncated towards zero.
 \end{itemdescr}
 
+\rSec2[numeric.sat]{Saturation arithmetic}
+
+\rSec3[numeric.sat.func]{Arithmetic functions}
+
+\pnum
+\begin{note}
+In the following descriptions, an arithmetic operation
+is performed as a mathematical operation with infinite range and then
+it is determined whether the mathematical result fits into the result type.
+\end{note}
+
+\indexlibraryglobal{add_sat}%
+\begin{itemdecl}
+template<class T>
+  constexpr T add_sat(T x, T y) noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\constraints
+\tcode{T} is a signed or unsigned integer type\iref{basic.fundamental}.
+
+\pnum
+\returns
+If \tcode{x + y} is representable as a value of type \tcode{T}, \tcode{x + y};
+otherwise, either the largest or smallest representable value of type \tcode{T},
+% FIXME: Do we define what is means for one value to be "closer" to another?
+whichever is closer to the value of \tcode{x + y}.
```

Comment:
This is the normal English use of the word. 130 is closer to 127 than to -128. 70000 is closer to 65535 than to 0.